onnxruntime/core/framework/op_kernel_context_internal.h

```diff
@@ -57,7 +58,8 @@ class OpKernelContextInternal : public OpKernelContext {
 
   const bool& GetTerminateFlag() const noexcept { return terminate_flag_; }
 
-  const onnxruntime::concurrency::ThreadPool* GetOperatorThreadPool() const { return session_state_.GetThreadPool(); }
+  _Ret_maybenull_ const onnxruntime::concurrency::ThreadPool* GetOperatorThreadPool() const { return session_state_.GetThreadPool(); }
+  _Ret_maybenull_ onnxruntime::concurrency::ThreadPool* GetOperatorThreadPool() { return session_state_.GetThreadPool(); }
 
```

Comment:
Shouldn't the Session always return a threadpool here (whether that eventually equates to an environment level one or a local one)?

Reply:
May need to give this more thought. The optimizer when constant folding creates an OpKernelContext not OpKernelContextInternal, so now any node that needs a threadpool is going to break.

That would suggest we move this method to OpKernelContext (but need to support a null pointer being returned for the threadpool as the optimizer shouldn't need to create a threadpool).

---
In reply to: [314568186](https://github.com/microsoft/onnxruntime/pull/1609#discussion_r314568186) [](ancestors = 314568186)

Reply:
see: _Ret_maybenull_.